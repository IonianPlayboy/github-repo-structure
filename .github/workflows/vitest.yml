name: Vitests Tests

on: push

jobs:
    test:
        runs-on: ubuntu-latest
        steps:
            - name: Check out code
              uses: actions/checkout@v2
            - name: Vitest run tests
              run: yarn && yarn test:unit:ci
