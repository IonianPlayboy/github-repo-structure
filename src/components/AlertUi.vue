<template>
	<div
		:class="colorStylesPerType[type].bg"
		class="rounded-md bg-gradient-to-r p-4 shadow-md"
		data-testid="alert"
	>
		<div class="flex">
			<div class="shrink-0">
				<component
					:is="currIcon"
					:class="colorStylesPerType[type].icon"
					class="h-5 w-5"
					aria-hidden="true"
				/>
			</div>
			<div class="ml-3">
				<h3
					:class="colorStylesPerType[type].title"
					class="text-sm font-medium"
				>
					<slot name="title" />
				</h3>
				<div
					v-if="slots?.default"
					:class="colorStylesPerType[type].text"
					class="mt-2 text-sm"
				>
					<p>
						<slot />
					</p>
				</div>
			</div>
		</div>
	</div>
</template>
<script setup lang="ts">
import { computed, useSlots } from "vue";
import {
	InformationCircleIcon,
	ExclamationIcon,
	XCircleIcon,
	CheckCircleIcon,
} from "@heroicons/vue/solid";

const props = defineProps<{
	type: "info" | "warning" | "error" | "success" | "disabled";
}>();

const slots = useSlots();

const iconsPerType = {
	info: InformationCircleIcon,
	warning: ExclamationIcon,
	error: XCircleIcon,
	success: CheckCircleIcon,
	disabled: InformationCircleIcon,
} as const;

const colorStylesPerType = {
	info: {
		bg: "from-blue-900/50 to-sky-900/50",
		icon: "text-blue-400",
		title: "text-blue-200",
		text: "text-blue-300",
	},
	warning: {
		bg: "from-yellow-900/50 to-amber-900/50",
		icon: "text-yellow-400",
		title: "text-yellow-200",
		text: "text-yellow-300",
	},
	error: {
		bg: "from-red-900/50 to-rose-900/50",
		icon: "text-red-400",
		title: "text-red-200",
		text: "text-red-300",
	},
	success: {
		bg: "from-green-900/50 to-emerald-900/50",
		icon: "text-green-400",
		title: "text-green-200",
		text: "text-green-300",
	},
	disabled: {
		bg: "from-slate-900/50 to-gray-900/50",
		icon: "text-slate-400",
		title: "text-slate-200",
		text: "text-slate-300",
	},
} as const;

const currIcon = computed(() => iconsPerType[props.type]);
</script>
