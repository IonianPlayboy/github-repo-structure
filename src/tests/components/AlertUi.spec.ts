import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import AlertUi from "@/components/AlertUi.vue";

describe("AlertUi", () => {
	it("renders properly", () => {
		const wrapper = mount(AlertUi, {
			props: { type: "info" },
			slots: {
				default: "Hello",
			},
		});
		expect(wrapper.text()).toContain("Hello");
		expect(wrapper.html()).toMatchSnapshot();
	});
	it("accepts a title and a text content", () => {
		const wrapper = mount(AlertUi, {
			props: { type: "info" },
			slots: {
				title: "Title !",
				default: "Hello this is text.",
			},
		});
		expect(wrapper.text()).toContain("Title !");
		expect(wrapper.text()).toContain("Hello this is text.");
		expect(wrapper.html()).toMatchSnapshot();
	});
	it("changes its styling according to its type", () => {
		const wrapperInfo = mount(AlertUi, {
			props: { type: "info" },
			slots: {
				default: "Hello bis",
			},
		});
		expect(
			wrapperInfo
				.classes()
				.some((className) => className.includes("blue"))
		).toBe(true);
		expect(wrapperInfo.html()).toMatchSnapshot();

		const wrapperError = mount(AlertUi, {
			props: { type: "error" },
			slots: {
				default: "Hello",
			},
		});
		expect(
			wrapperError
				.classes()
				.some((className) => className.includes("red"))
		).toBe(true);
		expect(wrapperError.html()).toMatchSnapshot();
	});
});
