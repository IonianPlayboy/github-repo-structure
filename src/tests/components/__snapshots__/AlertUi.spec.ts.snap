// Vitest Snapshot v1

exports[`AlertUi > accepts a title and a text content 1`] = `
"<div class=\\"from-blue-900/50 to-sky-900/50 rounded-md bg-gradient-to-r p-4 shadow-md\\" data-testid=\\"alert\\">
  <div class=\\"flex\\">
    <div class=\\"shrink-0\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" class=\\"text-blue-400 h-5 w-5\\">
        <path fill-rule=\\"evenodd\\" d=\\"M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z\\" clip-rule=\\"evenodd\\"></path>
      </svg></div>
    <div class=\\"ml-3\\">
      <h3 class=\\"text-blue-200 text-sm font-medium\\">Title !</h3>
      <div class=\\"text-blue-300 mt-2 text-sm\\">
        <p>Hello this is text.</p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`AlertUi > changes its styling according to its type 1`] = `
"<div class=\\"from-blue-900/50 to-sky-900/50 rounded-md bg-gradient-to-r p-4 shadow-md\\" data-testid=\\"alert\\">
  <div class=\\"flex\\">
    <div class=\\"shrink-0\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" class=\\"text-blue-400 h-5 w-5\\">
        <path fill-rule=\\"evenodd\\" d=\\"M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z\\" clip-rule=\\"evenodd\\"></path>
      </svg></div>
    <div class=\\"ml-3\\">
      <h3 class=\\"text-blue-200 text-sm font-medium\\"></h3>
      <div class=\\"text-blue-300 mt-2 text-sm\\">
        <p>Hello bis</p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`AlertUi > changes its styling according to its type 2`] = `
"<div class=\\"from-red-900/50 to-rose-900/50 rounded-md bg-gradient-to-r p-4 shadow-md\\" data-testid=\\"alert\\">
  <div class=\\"flex\\">
    <div class=\\"shrink-0\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" class=\\"text-red-400 h-5 w-5\\">
        <path fill-rule=\\"evenodd\\" d=\\"M10 18a8 8 0 100-16 8 8 0 000 16zM8.707 7.293a1 1 0 00-1.414 1.414L8.586 10l-1.293 1.293a1 1 0 101.414 1.414L10 11.414l1.293 1.293a1 1 0 001.414-1.414L11.414 10l1.293-1.293a1 1 0 00-1.414-1.414L10 8.586 8.707 7.293z\\" clip-rule=\\"evenodd\\"></path>
      </svg></div>
    <div class=\\"ml-3\\">
      <h3 class=\\"text-red-200 text-sm font-medium\\"></h3>
      <div class=\\"text-red-300 mt-2 text-sm\\">
        <p>Hello</p>
      </div>
    </div>
  </div>
</div>"
`;

exports[`AlertUi > renders properly 1`] = `
"<div class=\\"from-blue-900/50 to-sky-900/50 rounded-md bg-gradient-to-r p-4 shadow-md\\" data-testid=\\"alert\\">
  <div class=\\"flex\\">
    <div class=\\"shrink-0\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" class=\\"text-blue-400 h-5 w-5\\">
        <path fill-rule=\\"evenodd\\" d=\\"M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z\\" clip-rule=\\"evenodd\\"></path>
      </svg></div>
    <div class=\\"ml-3\\">
      <h3 class=\\"text-blue-200 text-sm font-medium\\"></h3>
      <div class=\\"text-blue-300 mt-2 text-sm\\">
        <p>Hello</p>
      </div>
    </div>
  </div>
</div>"
`;
